### PR TITLE
Merge tree service in master

### PR DESCRIPTION
**Pull request by fyrchik**
Only the last 2 commits are fresh.
Tree service must now be explicitly enabled via `tree.enabled` config field.

### CODE REVIEW

File: cmd/neofs-node/tree.go [hunk 1]
```diff
@@ -3,10 +3,15 @@ package main
 import (
 	"context"
 
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-node/config"
 	"github.com/nspcc-dev/neofs-node/pkg/services/tree"
 )
 
 func initTreeService(c *cfg) {
+	if !config.BoolSafe(c.appCfg.Sub("tree"), "enabled") {
```
Comment (carpawell) [inline cmd/neofs-node/tree.go:11]: just wanna mention: is it ok that admins gonna see "initializing tree service..." even if it's not enabled?

that question could also be applied to some other services, did not check

Comment (fyrchik) [reply cmd/neofs-node/tree.go:11]: No, added a log message.

**Review by carpawell — APPROVED** [review]

**Comment by fyrchik** [comment]
Updated only 2 fixes for the linter.

**Review by carpawell — APPROVED** [review]

**Comment by fyrchik** [comment]
Fixed conflicts with master.

**Review by carpawell — DISMISSED** [review]

**Review by carpawell — APPROVED** [review]

**Pull request merged**